DataFormats/PatCandidates/interface/PackedCandidate.h

```diff
@@ -68,12 +69,15 @@ namespace pat {
     }
 
     explicit PackedCandidate( const PolarLorentzVector &p4, const Point &vtx,
-                              float phiAtVtx, int pdgId,
+                              float trkPt,float etaAtVtx,float phiAtVtx,int pdgId,
                               const reco::VertexRefProd &pvRefProd,
                               reco::VertexRef::key_type pvRefKey) :
       packedPuppiweight_(0), packedPuppiweightNoLepDiff_(0), hcalFraction_(0),
       p4_( new PolarLorentzVector(p4) ), p4c_( new LorentzVector(*p4_)), 
-      vertex_( new Point(vtx) ), dphi_(reco::deltaPhi(phiAtVtx,p4_.load()->phi())), 
+      vertex_( new Point(vtx) ), 
+      dphi_(reco::deltaPhi(phiAtVtx,p4_.load()->phi())), 
+      deta_(std::abs(etaAtVtx-p4_.load()->eta())>=kMinDEtaToStore_ ? etaAtVtx-p4_.load()->eta() : 0.), 
+      dtrkpt_(std::abs(trkPt-p4_.load()->pt())>=kMinDTrkPtToStore_ ? trkPt-p4_.load()->pt() : 0.),
```

Comment:
shouldn't this be made fractional instead of absolute?
almost all tracks with pt>10 GeV will have non-zero here ... OTOH, the fraction of these in the total is small.
So, not very essential to change.